### DISCUSSION TITLE
Is it possible to send custom RTCP packets

### DISCUSSION THREAD

**Discussion by chenhao523** [bluenviron/mediamtx#3111] [Questions]
### Question

Excuse me, do you know if there's an API available for sending customized RTCP packets? If so, could you also suggest some good practices for doing this?

**Comment by aler9** [comment]
Hello, there is not an API and it would be troublesome to implement it, since there is a wide variety of RTCP packets, each with its own meaning. In my opinion you'd better hack the server source code (if you're interested into RTCP for WebRTC) or the source code of gortsplib (if you're interested into RTCP for RTSP) and add a custom logic there. Both are based on the [pion/rtcp](https://github.com/pion/rtcp) library that allows to encode and decode RTCP packets.

[a comment by an automated account is not shown]
